### PR TITLE
Multi stage NN pipeline for custom nets

### PR DESCRIPTION
**Pull request by daniilpastukhov**

### CODE REVIEW

**Comment by daniilpastukhov** [comment]
Added support for rotated detections (e.g., OCR texts)

File: depthai_sdk/src/depthai_sdk/components/nn_component.py [hunk 1]
```diff
@@ -126,37 +136,63 @@ def _update_device_info(self, pipeline: dai.Pipeline, device: dai.Device, versio
...
+                self.image_manip.setResize(*self._size)
+                # self.image_manip.setWaitForConfigInput(True)
+                # self.image_manip.inputConfig.setReusePreviousMessage(True)
+
+                self.image_manip.setMaxOutputFrameSize(self._size[0] * self._size[1] * 3)
+
+                self.x_in = pipeline.createXLinkIn()
+                self.x_in.setStreamName("input_queue")
+                self.x_in.setMaxDataSize(frame_size[0] * frame_size[1] * 3)
+                self.x_in.out.link(self.image_manip.inputImage)
```
Comment (Erol444) [inline depthai_sdk/src/depthai_sdk/components/nn_component.py:187]: Maybe add TODO for the future that we could also just queue high-res img on the device (script node) then send config from host and Script node would send queued up Image + config to the ImageManip

File: depthai_sdk/src/depthai_sdk/components/nn_component.py [hunk 2]
```diff
@@ -191,33 +227,38 @@ def _parse_model(self, model):
...
                 self._parse_config(model)
         else:  # SDK supported model
             models = getSupportedModels(printModels=False)
-            if str(model) not in models:
+            zoo_models = blobconverter.zoo_list()
+
+            if str(model) in models:
+                model = models[str(model)] / 'config.json'
+                self._parse_config(model)
+            elif str(model) in zoo_models:
```
Comment (Erol444) [inline depthai_sdk/src/depthai_sdk/components/nn_component.py:240]: The problem with going with blobconverter is that we don't know anything about the model; label map, decoding logic, metadata (eg yolo), etc. so we have to make sure to throw useful error if these aren't provided to SDK via some other api.

File: depthai_sdk/src/depthai_sdk/nn_models/deeplabv3/model.yml [hunk 3]
```diff
@@ -3,11 +3,11 @@ description: >-
 documentation: https://github.com/PINTO0309/PINTO_model_zoo/blob/master/026_mobile-deeplabv3-plus/01_float32/download.sh
 task_type: semantic_segmentation
 files:
-  - name: FP16/deeplabv3p_person.xml
+  - name: FP16/deeplabv3.xml
```
Comment (Erol444) [inline depthai_sdk/src/depthai_sdk/nn_models/deeplabv3/model.yml:6]: is this correct?

**Review by Erol444 — COMMENTED** [review]
Also next time please have syntax refactors in other PR, and for some reason roboflow integration commits came into this PR? There was a lot of unnecessary checking/scrolling:)

File: depthai_sdk/src/depthai_sdk/components/nn_component.py (hunk 2 above)
Comment (daniilpastukhov) [reply depthai_sdk/src/depthai_sdk/components/nn_component.py:241]: Added warning

File: depthai_sdk/src/depthai_sdk/components/nn_component.py (hunk 1 above)
Comment (daniilpastukhov) [reply depthai_sdk/src/depthai_sdk/components/nn_component.py:188]: Added TODO

File: depthai_sdk/src/depthai_sdk/nn_models/deeplabv3/model.yml (hunk 3 above)
Comment (daniilpastukhov) [reply depthai_sdk/src/depthai_sdk/nn_models/deeplabv3/model.yml]: Rollbacked, most likely changed it when used PyCharm refactoring, thanks

**Review by Erol444 — APPROVED** [review]
lgtm if points from previous review were addressed

**Pull request merged**